source/slang/hlsl.meta.slang

```diff
@@ -4589,7 +4592,7 @@ T exp2(T x)
         }
         else
         {
-            float xf = __slang_noop_cast<float>(x);
+            float xf = __floatCast<float>(x);
```

Comment:
There is already a `__realCast` defined in core.meta.slang that does exactly this, so we can just use that. Sorry I didn't realize that earlier.